avro/src/test/java/org/apache/apex/malhar/avro/AvroFileToPojoModuleTest.java

```diff
@@ -0,0 +1,269 @@
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileContext;
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.api.Attribute;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.Context.PortContext;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.StreamingApplication;
+import com.datatorrent.lib.helper.TestPortContext;
+import com.datatorrent.lib.io.ConsoleOutputOperator;
+
+import static com.datatorrent.lib.helper.OperatorContextTestHelper.mockOperatorContext;
+
+/**
+ * <p>
+ * In this class the emitTuples method is called twice to process the first
+ * input, since on begin window 0 the operator is setup & stream is initialized.
+ * The platform calls the emitTuples method in the successive windows
+ * </p>
+ */
+public class AvroFileToPojoModuleTest
+{
+
+  private static final String AVRO_SCHEMA = "{\"namespace\":\"abc\"," + ""
+      + "\"type\":\"record\",\"doc\":\"Order schema\"," + "\"name\":\"Order\",\"fields\":[{\"name\":\"orderId\","
+      + "\"type\": \"long\"}," + "{\"name\":\"customerId\",\"type\": \"int\"},"
+      + "{\"name\":\"total\",\"type\": \"double\"}," + "{\"name\":\"customerName\",\"type\": \"string\"}]}";
+
+  private static final String FILENAME = "/tmp/simpleorder.avro";
+  private static final String OTHER_FILE = "/tmp/simpleorder2.avro";
+
+  AvroFileToPojoModule avroFileToPojoModule = new AvroFileToPojoModule();
+
+  private List<GenericRecord> recordList = null;
+
+  public static class TestMeta extends TestWatcher
+  {
+    public String dir = null;
+    OperatorContext context;
+    PortContext portContext;
+
+    @Override
+    protected void starting(org.junit.runner.Description description)
+    {
+      String methodName = description.getMethodName();
+      String className = description.getClassName();
+      this.dir = "target/" + className + "/" + methodName;
+      Attribute.AttributeMap attributes = new Attribute.AttributeMap.DefaultAttributeMap();
+      attributes.put(Context.DAGContext.APPLICATION_PATH, dir);
+      context = mockOperatorContext(1, attributes);
+      Attribute.AttributeMap portAttributes = new Attribute.AttributeMap.DefaultAttributeMap();
+      portAttributes.put(Context.PortContext.TUPLE_CLASS, SimpleOrder.class);
+      portContext = new TestPortContext(portAttributes);
+    }
+
+    @Override
+    protected void finished(Description description)
+    {
+      try {
+        FileUtils.deleteDirectory(new File("target/" + description.getClassName()));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Rule
+  public TestMeta testMeta = new TestMeta();
+
+  private void createAvroInput(int cnt)
```

Comment:
Let us not worry about refactoring in this jira. You can create a separate jira.